examples/08_deduplication.py

```diff
@@ -0,0 +1,163 @@
+# As a data scientist, our job is to analyze the data from a hospital ward.
+# We notice that most of the cases involve the prescription of one of three different medications - "Contrivan", "Genericon", or "Zipholan".
+# However, data entry is manual and - either because the prescribing doctor's handwriting was hard to decipher or due to mistakes during data input - there are multiple spelling mistakes for these three medications.
+#
+# Let's generate some example data that demonstrate this.
+
+
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+
+
+def generate_example_data(examples, entries_per_example, prob_mistake_per_letter):
+    """Helper function to generate data consisting of multiple entries per example.
+    Characters are misspelled with probability `prob_mistake_per_letter`"""
+    import string
+
+    data = []
+    for example, n_ex in zip(examples, entries_per_example):
+        len_ex = len(example)
+        # generate a 2D array of chars of size (n_ex, len_ex)
+        str_as_list = np.array([list(example)] * n_ex)
+        # randomly choose which characters are misspelled
+        idxes = np.where(
+            np.random.random(len(example[0]) * n_ex) < prob_mistake_per_letter
+        )[0]
+        # and randomly pick with which character to replace
+        replacements = [
+            string.ascii_lowercase[i]
+            for i in np.random.choice(np.arange(26), len(idxes)).astype(int)
+        ]
+        # introduce spelling mistakes at right examples and right char locations per example
+        str_as_list[idxes // len_ex, idxes % len_ex] = replacements
+        # go back to 1d array of strings
+        data.append(np.ascontiguousarray(str_as_list).view(f"U{len_ex}").ravel())
+    return np.concatenate(data)
+
+
+np.random.seed(123)
+
+# our three medication names
+medications = ["Contrivan", "Genericon", "Zipholan"]
+entries_per_medications = [500, 100, 1500]
+
+# 5% probability of a typo per letter
+prob_mistake_per_letter = 0.05
+
+data = generate_example_data(
+    medications, entries_per_medications, prob_mistake_per_letter
+)
+# we extract the unique medication names in the data & how often they appear
+unique_examples, counts = np.unique(data, return_counts=True)
+# and build a series out of them
+ex_series = pd.Series(counts, index=unique_examples)
+
+
+# ## Visualize the data
+
+
+ex_series.plot.bar(figsize=(20, 10))
+_ = plt.xlabel("Medication name")
+_ = plt.ylabel("Counts")
+
+
+# We can now see clearly the structure of the data: The three original medications are the most common ones, however there are many spelling mistakes and hence many slight variations of the names of the original medications.
+#
+# The idea is to use the fact that the string-distance of each misspelled medication name will be closest to either the correctly or incorrectly spelled orginal medication name - and therefore form clusters.
+
+
+
+from dirty_cat.deduplicate import deduplicate, compute_ngram_distance
```

Comment:
I would import the function deduplicate from the main `__init__` of dirty-cat. This is where we have been putting all our user-facing functions and objects so far.